Maadeyo9/wundertest
# Note-API with CRUD functionality (json)

Test Cases:

1. Add notes
Add 3 notes:
* call POST http://localhost/notes/add  with json object { "title": "one", "text": "content 1" }
* call POST http://localhost/notes/add with json object { "title": "two", "text": "content 2" }
* call POST http://localhost/notes/add with json object { "title": "three", "text": "content 3" }
Verify notes are created in db:
* Open phpmyadmin, check contents of Note table and verify it has 3 notes with correct contents

2. Edit notes
You can edit a note by calling a PUT to localhost/notes/{id}/edit 
with json object { "title": "one", "text": "content" }
This can also be verified via checking the Note table on phpmyadmin.

3. List all notes:
* Call GET on http://localhost/notes/ 
Verify notes are in ordered by creation time. 

4. Showing options
You can input options by adding "limit" or "reversed" to the url, for example http://localhost/notes/?reversed
Limit applies a limit to the amount of notes shown, and "reversed" reverses the order. By default the notes are sorted by their time of creation, with the newest one being on the top.

5. Show specific note

You can show a specific note by calling a GET on http://localhost/notes/{id} where the {id} is the number of the note created.

6. Delete notes:

You can delete created notes by calling a DELETE on https://localhost/notes/{id}/delete.
The deletion can be verified by looking at the contents of the Note table on phpmyadmin.
